### PR TITLE
Update documentation

### LINKED ISSUE #96: Update documentation
Write proper documentation for bindgen using Paradox

### PR DESCRIPTION
**Pull request by kornilova-l**
What will be a proper way to deploy documentation?
One option is to use github pages but pushing generated htmls and other files to the repo does not look good to me.
Closes #96

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -88,3 +88,13 @@ lazy val samples = project
...
 
 lazy val tools = project in file("tools")
+
+lazy val docs = project
+  .in(file("docs"))
+  .enablePlugins(ParadoxPlugin)
+  .settings(
+    paradoxTheme := Some(builtinParadoxTheme("generic")),
+    paradoxProperties in Compile ++= Map(
+      "github.base_url" -> "https://github.com/kornilova-l/scala-native-bindgen/tree/master/"
```
Comment (kornilova-l) [inline build.sbt]: It should be something like `s"https://github.com/kornilova-l/scala-native-bindgen/tree/v${version.value}"`
But what will be the way to release new version then?
* manually change `version := "0.2-SNAPSHOT"` to `version := "0.2"`
* generate docs
* release
* manually change `version := "0.2"` to `version := "0.3-SNAPSHOT"`

Is there a way to automate this?

**Comment by kornilova-l** [comment]
I found out that it is also possible to push generated docs to `gh-pages` branch

**Comment by jonas** [comment]
What was the reason to switch from Paradox to GitBook? Was it problem with configuration or something else? I personally think Paradox is a better fit since it has better indexing capabilities and because GitBook seems to have gone closed source.

For publishing we can use `sbt-site` with `sbt-ghpages`: https://www.scala-sbt.org/sbt-site/publishing.html#publishing-to-github-pages-using-sbt-ghpages

**Comment by kornilova-l** [comment]
No particular reason except that GitBook has github-like rendering, I moved back to Paradox
`sbt-ghpages` works :+1:

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 target/
 scripts/.coursier
 scripts/.scalafmt-*
+/docs/_book/
```
Comment (jonas) [inline .gitignore]: This is no longer needed.

**Pull request merged**